test/python/compiler/test_transpiler.py

```diff
@@ -1756,21 +1756,29 @@ def test_no_infinite_loop(self, optimization_level):
             seed_transpiler=42,
         )
 
-        # Expect a -pi/2 global phase for the U3 to RZ/SX conversion, and
-        # a -0.5 * theta phase for RZ to P twice, once at theta, and once at 3 pi
-        # for the second and third RZ gates in the U3 decomposition.
-        expected = QuantumCircuit(
-            1, global_phase=-np.pi / 2 - 0.5 * (-0.2 + np.pi) - 0.5 * 3 * np.pi
-        )
-        expected.p(-np.pi, 0)
-        expected.sx(0)
-        expected.p(np.pi - 0.2, 0)
-        expected.sx(0)
+        if optimization_level == 1:
+            # Expect a -pi/2 global phase for the U3 to RZ/SX conversion, and
+            # a -0.5 * theta phase for RZ to P twice, once at theta, and once at 3 pi
+            # for the second and third RZ gates in the U3 decomposition.
+            expected = QuantumCircuit(
+                1, global_phase=-np.pi / 2 - 0.5 * (-0.2 + np.pi) - 0.5 * 3 * np.pi
+            )
+            expected.p(-np.pi, 0)
+            expected.sx(0)
+            expected.p(np.pi - 0.2, 0)
+            expected.sx(0)
+        else:
+            expected = QuantumCircuit(1, global_phase=(15 * np.pi - 1) / 10)
```

Comment:
This is slightly unexpected, I guess at some point a gate is now removed and `Optimize1qGatesDecomposition` optimizes the sequence differently? Or why do you think this happens?

Reply:
That was my assumption, I don't remember the exact details though. Let me run it with a callback and see what's going on exactly.

Reply:
Looking through each pass it looks like `RemoveIdentityEquivalent` reduces changes:
```
global phase: 4.6124
   ┌──────┐┌────┐┌───────────┐┌────┐┌───────┐
q: ┤ P(0) ├┤ √X ├┤ P(3.3416) ├┤ √X ├┤ P(3π) ├
   └──────┘└────┘└───────────┘└────┘└───────┘
   ```
   
   to:
```
global phase: 4.6124
   ┌────┐┌───────────┐┌────┐┌───────┐
q: ┤ √X ├┤ P(3.3416) ├┤ √X ├┤ P(3π) ├
   └────┘└───────────┘└────┘└───────┘
```

This changes the output of `Optimize1qGatesDecomposition` which runs immediately after from:
```
global phase: 4.8124
   ┌───────┐┌────┐┌───────────┐┌────┐
q: ┤ P(-π) ├┤ √X ├┤ P(2.9416) ├┤ √X ├
   └───────┘└────┘└───────────┘└────┘
```

which is the output on calling the pass on the circuit with `P(0)`

to:

```
global phase: 4.6124
   ┌────┐┌───────────┐┌────┐┌───────┐
q: ┤ √X ├┤ P(3.3416) ├┤ √X ├┤ P(3π) ├
   └────┘└───────────┘└────┘└───────┘
```
which is the pass's output with `P(0)` removed. What this is coming down to is the preference for `Optimize1qGatesDecomposition` to defer to the input circuit if it doesn't think the synthesis output is any better. Since the gate counts of either output are the same, the pass views them as equivalent (in the absence of error rates in the target) and prefer to not modify the circuit instead of use the synthesis output. While in the case of without the removal pass being run there is an extra 1q gate in the sequence the pass views the synthesis output as being more efficient because it has 1 fewer gate.

Reply:
Ahh that makes sense -- thanks for checking, better to be sure! 🙂